src/graphics.c

```diff
@@ -616,10 +632,25 @@ void graphics_fill_screen( display_context_t disp, uint32_t c )
 }
 
 /**
- * @brief Draw a character to the screen using the built-in font
+ * @brief Set the current font. Should be set before using any of the draw function.
  *
- * @todo Need a set font mechanism.
- * @todo Need to look up the width and height from the font instead of assuming it is 8x8.
+ * @param[in] font
+ *        Sprite font to be used. Should be using a standard font sprite (containing every ASCII
+ * character) and with a transparent background, and be loaded on mksprite with width and height
+ * of 16.
```

Comment:
Yeah, I was trying to use this as an opportunity to explain to the user how to use it. But since I created an example it became just more confusing.

And honestly the size doesn't matter, as long as the sprite imported has enough slices to use (I even used a 16x8 on the example).

Should it be better to just point the user toward the example to see how to use it instead of trying to come up with a good explanation?

Reply:
Pointing to the example is always good, but I think also the example shows "one possible way". I think the documentation should state the exact requirements for the font bitmap, so that users can evaluate their options. I don't think a single example can capture all possibilities